### PR TITLE
docs: rephrase plugin configuration in docs

### PR DESCRIPTION
**Pull request by Secrus**
# Pull Request Check List

Resolves: #6121

- [ ] Added **tests** for changed code. (**does not include code changes**)
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ This will help keep the performances of Poetry good.
 {{% /note %}}
 
 The plugin also must be declared in the `pyproject.toml` file of the plugin package
-as an `application.plugin` plugin:
+as an `poetry.application.plugin` plugin:
```
Comment (neersighted) [inline docs/plugins.md:122]:
```suggestion
as a `poetry.application.plugin` plugin:
```

File: docs/pyproject.md [hunk 2]
```diff
@@ -410,16 +410,21 @@ Dependencies listed in [dependency groups]({{< relref "managing-dependencies#dep
 
 ## `plugins`
 
-Poetry supports arbitrary plugins which work similarly to
-[setuptools entry points](http://setuptools.readthedocs.io/en/latest/setuptools.html).
-To match the example in the setuptools documentation, you would use the following:
+Poetry supports arbitrary plugins which work similarly to `setuptools` entry points.
```
Comment (neersighted) [inline docs/pyproject.md:413]:
```suggestion
Poetry supports arbitrary plugins, which are exposed as the ecosystem-standard [entry points](https://packaging.python.org/en/latest/specifications/entry-points/) and discoverable using `importlib.metadata`. This is similar to (and compatible with) the entry points feature of `setuptools`.
```

File: docs/pyproject.md [hunk 3]
```diff
@@ -410,16 +410,21 @@ Dependencies listed in [dependency groups]({{< relref "managing-dependencies#dep
...
-[tool.poetry.plugins."blogtool.parsers"]
-".rst" = "some_module:SomeClass"
+[tool.poetry.plugins."A"]
+"B" = "C:D"
 ```
+Which are:
+
+- `A` - type of the plugin, for example `poetry.plugin` or `flake8.extension`
+- `B` - name of the plugin
+- `C` - pythonic path to module containing plugin class
```
Comment (neersighted) [inline docs/pyproject.md:426]:
```suggestion
- `C` - python module import path
```

File: docs/pyproject.md [hunk 4]
```diff
@@ -410,16 +410,21 @@ Dependencies listed in [dependency groups]({{< relref "managing-dependencies#dep
...
+- `D` - name of plugin class
```
Comment (neersighted) [inline docs/pyproject.md:427]:
```suggestion
- `D` - the entry point of the plugin (a function or class)
```

**Review by neersighted — CHANGES_REQUESTED** [review]

File: docs/pyproject.md [hunk 5]
```diff
@@ -410,16 +410,21 @@ Dependencies listed in [dependency groups]({{< relref "managing-dependencies#dep
...
-".rst" = "some_module:SomeClass"
+[tool.poetry.plugins."A"]
+"B" = "C:D"
 ```
+Which are:
+
+- `A` - type of the plugin, for example `poetry.plugin` or `flake8.extension`
+- `B` - name of the plugin
+- `C` - python module import path
+- `D` - the entry point of the plugin (a function or class)
```
Comment (mkniewallner) [inline docs/pyproject.md:427]: IMO, having an example usage additionally to the template and details would give even more insights to user.

What about showcasing our own export plugin, as a real life example?
```toml
[tool.poetry.plugins."poetry.application.plugin"]
export = "poetry_plugin_export.plugins:ExportApplicationPlugin"
```

Comment (Secrus) [reply docs/pyproject.md:427]: Good idea, I have added the example.

**Review by mkniewallner — APPROVED** [review]

**Pull request merged**